src/hooks/useEnvironmentSelection.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { NativeEventEmitter } from 'react-native';
+
+const eventEmitter = new NativeEventEmitter({
+  addListener: () => {},
+  removeListeners: () => {},
+});
+
+const environmentToggledEvent = 'environmentToggledEvent';
+
+export const emitEnvironmentToggled = (value: string) =>
+  eventEmitter.emit(environmentToggledEvent, value);
+
+export const useEnvironmentSelection = () => {
+  const [usePrimaryEnvironment, setUsePrimaryEnvironment] = useState(true);
```

Comment:
Should this check async storage? If the app is using secondary, I think this would be true on load.

Reply:
It will get automatically overwritten when the useEnvironmentSelection hook loads so it may appear in the primary position for a split second. It was originally reading from async storage but I decided that a single read from the hook would be sufficient for synchronizing the selection.